modules/fluid_properties/src/base/FluidPropertiesApp.C

```diff
@@ -24,71 +24,62 @@ registerKnownLabel("FluidPropertiesApp");
 
 FluidPropertiesApp::FluidPropertiesApp(InputParameters parameters) : MooseApp(parameters)
 {
-  Moose::registerObjects(_factory);
-  FluidPropertiesApp::registerObjects(_factory);
-
-  Moose::associateSyntax(_syntax, _action_factory);
-  FluidPropertiesApp::associateSyntax(_syntax, _action_factory);
-
-  Moose::registerExecFlags(_factory);
-  FluidPropertiesApp::registerExecFlags(_factory);
+  FluidPropertiesApp::registerAll(_factory, _action_factory, _syntax);
 }
 
 FluidPropertiesApp::~FluidPropertiesApp() {}
 
-// External entry point for dynamic application loading
-extern "C" void
-FluidPropertiesApp__registerApps()
-{
-  FluidPropertiesApp::registerApps();
-}
-
 void
 FluidPropertiesApp::registerApps()
 {
   registerApp(FluidPropertiesApp);
 }
 
-// External entry point for dynamic object registration
-extern "C" void
-FluidPropertiesApp__registerObjects(Factory & factory)
+static void
+associateSyntaxInner(Syntax & syntax, ActionFactory & /*action_factory*/)
 {
-  FluidPropertiesApp::registerObjects(factory);
+  registerSyntaxTask(
+      "AddFluidPropertiesAction", "Modules/FluidProperties/*", "add_fluid_properties");
```

Comment:
RELAP-7 and pronghorn register this syntax under `FluidProperties/*`, i.e. without the `Modules` prefix. The `registerAll` approach is nice, but we would need a way how to achieve the custom syntax path.  Otherwise when we start using it it will break all our input files.

Originally, we would skip the syntax registration and would have our own implementation that would do the right thing.  We would be fine with the `registerAll` approach as long as we could tell (may be upfront), that there will be some remapping going on. Or, any other simple solution...

Reply:
Calling:

```c++
syntax.replaceActionSyntax(AddFluidPropertiesAction,
                            "FluidProperties/*",
                            "add_fluid_properties",
                            __FILE__,
                            __LINE__);
```
 after the registerAll call should do the trick.

Reply:
Cool! I will test it in RELAP. Thanks!

Reply:
This worked. Just a small correction, the first parameter for `replaceActionSyntax` is `std::string`, so it should be "AddFluidPropertiesAction" - in case somebody finds this in future.